go/plugins/googleai/googleai.go

```diff
@@ -128,30 +130,31 @@ func Init(ctx context.Context, g *genkit.Genkit, cfg *Config) (err error) {
 		}
 	}
 
-	opts := append([]option.ClientOption{
-		option.WithAPIKey(apiKey),
-		genai.WithClientInfo("genkit-go", internal.Version),
-	},
-		cfg.ClientOptions...,
-	)
-	client, err := genai.NewClient(ctx, opts...)
+	xGoogApiClientHeader := http.CanonicalHeaderKey("x-goog-api-client")
+	client, err := genai.NewClient(ctx, &genai.ClientConfig{
+		APIKey:  apiKey,
+		Backend: genai.BackendGeminiAPI,
+		HTTPOptions: genai.HTTPOptions{
+			Headers: http.Header{
+				xGoogApiClientHeader: {fmt.Sprintf("genkit-go/%s", internal.Version)},
```

Comment:
Define a `internal.GenkitClientHeader` with this value and use that instead please.